transaction/core/src/amount/mod.rs

```diff
@@ -81,12 +115,92 @@ impl Amount {
             return Err(AmountError::InconsistentCommitment);
         }
 
-        Ok((value, blinding))
+        Ok((amount_data, blinding))
+    }
+
+    /// Compute the crc32 of the compressed commitment
+    pub fn commitment_crc32(&self) -> u32 {
+        Self::compute_commitment_crc32(&self.commitment)
+    }
+
+    /// Recovers an Amount from only the masked value and masked_token_id, and
+    /// shared secret.
+    ///
+    /// Note: This fails and produces gibberish if the shared secret is wrong.
+    ///
+    /// * You should confirm by checking against the real commitment, or the the
+    ///   crc32 of commitment.
+    ///
+    /// Arguments:
+    /// * masked_value: u64
+    /// * masked_token_id: &[u8], either 0 or 4 bytes
+    /// * shared_secret: The shared secret curve point
+    ///
+    /// Returns:
+    /// * Amount
+    /// * AmountData (token id and value)
+    /// or
+    /// * An amount error
+    pub fn reconstruct(
```

Comment:
@voloshyn i noticed that in a lot of the SDKs, this function is basically being reimplemented from scratch using `get_value_mask` etc, i think it will be more maintainable if we try to use this version instead